### PR TITLE
Reduces lag caused by bags

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
fix: Storage bags should now cause less lag when picking up large amounts of items.
fix: Storage bags now don't send an error message for every single item they fail to pick up.
/:cl:

Storage bags now stop looping if they're full, which is especially useful if a miner walks on a tile containing hundereds of stacked ores repeatedly. Also it won't spam error messages anymore, there's no need to say "This item cannot be put in this kind of bag" a dozen times.
Lag will still happen if someone tries to use a bag on a large amount of non-bag items.

### CODE REVIEW

**Comment by RemieRichards** [comment]
If you want to further help deal with the lag, make ```can_hold``` a typecache, because:
https://github.com/tgstation/tgstation/blob/037aed94f7c4f0a4b59b1835bd71c3e7207e4767/code/game/objects/items/weapons/storage/storage.dm#L264 is disgusting.

**Comment by coiax** [comment]
Also, while you're here, can you fix the spam of *pickup noise* if you pick up a bunch of stuff

**Comment by XDTM** [comment]
This should do both what @coiax and @RemieRichards asked.

File: code/game/objects/items/weapons/storage/storage.dm [hunk 1]
```diff
@@ -259,18 +259,12 @@
...
-				break
-		if(!ok)
+		if(!is_type_in_typecache(W, can_hold))
 			if(!stop_messages)
 				usr << "<span class='warning'>[src] cannot hold [W]!</span>"
 			return 0
-
-	for(var/A in cant_hold) //Check for specific items which this container can't hold.
-		if(istype(W, A))
+	else
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/storage/storage.dm]: bad, this changes the code flow, this means cant hold is completely overriden by can hold, even for items not shared between lists.

Comment (XDTM) [inline code/game/objects/items/weapons/storage/storage.dm]: If you look at the comments in the cant_hold definition, it is only supposed to work when can_hold is empty, i was enforcing that. I'll swap it back if you want to, of course.

Comment (RemieRichards) [inline code/game/objects/items/weapons/storage/storage.dm]: That comment has been unenforced for ages, thus it shouldn't *be* enforced without discussion.
So yes, revert your change in that sense.

**Comment by XDTM** [comment]
Done

**Pull request merged**